src/wallet/wallet2.cpp

```diff
@@ -355,7 +357,7 @@ std::unique_ptr<tools::wallet2> generate_from_json(const std::string& json_file,
       return false;
     }
 
-    wallet.reset(new tools::wallet2(testnet, restricted));
+    wallet.reset(make_basic(vm, opts).release());
```

Comment:
Assignment actually works in this situation; `wallet = make_basic(vm_opts);`. The current version should have identical behavior FWIW.